e2e_samples/fabric_dataops_sample/config/fabric_environment/otel_monitor_invoker.py

```diff
@@ -0,0 +1,133 @@
+#    pre-configured to send data to azure-monitor.
+# ref: https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=python
+# ```
+# from azure.monitor.opentelemetry import configure_azure_monitor
+
+# # Configure the Distro to authenticate with Azure Monitor - without managed identity
+# configure_azure_monitor(
+#     connection_string="your-connection-string"
+# )
+
+# # using a managed identity credential.
+# configure_azure_monitor(
+#     connection_string="your-connection-string",
+#     credential=ManagedIdentityCredential(),
+# )
+# ```
+# - https://learn.microsoft.com/en-us/python/api/overview/azure/monitor-opentelemetry-exporter-readme?view=azure-python-preview
+# - https://learn.microsoft.com/en-us/python/api/overview/azure/monitor-opentelemetry-exporter-readme?view=azure-python-preview#examples
+
+import logging
+
+# from azure.monitor.opentelemetry import configure_azure_monitor # We are using the advanced
+#    configs shown below using AzureMonitor*Exporter.
+from azure.monitor.opentelemetry.exporter import (
+    AzureMonitorLogExporter,
+    AzureMonitorMetricExporter,
+    AzureMonitorTraceExporter,
+)
+
+# metrics
+# traces
+from opentelemetry import metrics, trace
+
+# logs
+from opentelemetry.sdk._logs import LoggerProvider, LoggingHandler
+from opentelemetry.sdk._logs.export import BatchLogRecordProcessor
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+logging.basicConfig(level=logging.INFO)
+
+
+class OpenTelemetryAppInsightsExporter:
+    def __init__(self, conn_string: str) -> None:
+        """
+        Initializes the OpenTelemetryExporter class.
+
+        Args:
+            conn_string (str): Azure AppInsights connection string.
+        """
+        self.conn_string = conn_string
+
+        return None
+
+    def get_otel_tracer(self, trace_resource_attributes: dict, tracer_name: str = __name__) -> object:
+        """
+        Creates and returns an OpenTelemetry tracer object.
+
+        Args:
+            trace_resource_attributes (dict): The OpenTelemetry resource attributes in dictionary format
+            tracer_name (str): The name of the tracer. Default is __name__.
+        Returns:
+            tracer: OpenTelemetry tracer object
+        """
+        resource = Resource(attributes=trace_resource_attributes)
+        tracer_provider = TracerProvider(resource=resource)
+        # Exporter to send data to AppInsights
+        trace_exporter = AzureMonitorTraceExporter(connection_string=self.conn_string)
+        span_processor = BatchSpanProcessor(trace_exporter)
+        tracer_provider.add_span_processor(span_processor)
+        tracer = trace.get_tracer(tracer_name, tracer_provider=tracer_provider)
```

Comment:
Hi @sreedhar-guda, it's tiny thing but let me ask.
Do you have any intention here why not setting `tracer_provider` globally like `trace.set_tracer_provider(tracer_provider)`.
The reason why I'm asking this is because this code will have a following error (but it's not blocker) during exporting.
```
ERROR:azure.monitor.opentelemetry.exporter.export.trace._exporter:Failed to derive Resource from Tracer Provider: 'ProxyTracerProvider' object has no attribute 'resource'
```
(Reference:[ azure monitor otel github](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/monitor/azure-monitor-opentelemetry-exporter#tracing-1:~:text=trace.set_tracer_provider(tracer_provider)) looks globally setting tracer_provider

Reply:
@maniSbindra , do you know the above details?